Assets/Scripts/Enemies/Enemy.cs

```diff
@@ -5,38 +5,48 @@
 
 public class Enemy : MonoBehaviour
 {
-    Animator anim;
-    Transform target;
+    public IEnemyState currentState;
+
+    //public PlayerStats playerStats;   // PlayerStats script currently unavailable
+   
+    // Placeholder player stats
+    public float playerHP = 100;
+
+    public Animator anim;
+    public Transform player;
 
     public int hp;
     public int damage;
     public float speed;
-    float detectionRange;
-    float attackRange;
+    public float detectionRange;
+    public float attackRange;
     bool isDead = false;
 
-    void Start()
+    protected void Start()
     {
-        
+        anim = GetComponent<Animator>();
```

Comment:
Add RequiresComponent<Animator> to the top of your class if you assume this will always have one.